### PR TITLE
chore(roll): roll Playwright to 1.48.0-beta-1728034490000

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-python/issues/2580

- [x] Port tests over to sync test suite

### CODE REVIEW

File: tests/server.py [hunk 1]
```diff
@@ -280,6 +287,21 @@ def listen(self, factory: http.HTTPFactory) -> None:
 
 
 class WebSocketProtocol(WebSocketServerProtocol):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.events = EventEmitter()
```
Comment (mxschmitt) [inline tests/server.py:292]: this basically rewrites onMessage and onClose to an easier to consume pyee event emitter.

File: playwright/sync_api/_generated.py [hunk 2]
```diff
@@ -14416,11 +14676,10 @@ def launch_persistent_context(
             `passphrase` property should be provided if the certificate is encrypted. The `origin` property should be provided
             with an exact match to the request origin that the certificate is valid for.
 
-            **NOTE** Using Client Certificates in combination with Proxy Servers is not supported.
```
Comment (mxschmitt) [inline playwright/sync_api/_generated.py:14419]: Notes seem to not get rendered anymore.

File: playwright/_impl/_fetch.py [hunk 3]
```diff
@@ -405,7 +404,8 @@ async def _inner_fetch(
             "fetch",
             {
                 "url": url,
-                "params": params_to_protocol(params),
+                "params": object_to_array(params) if isinstance(params, dict) else None,
+                "encodedParams": params if isinstance(params, str) else None,
```
Comment (dgozman) [inline playwright/_impl/_fetch.py:408]: I haven't seen this change in .net or java roll for some reason. Could you please double check?

File: playwright/_impl/_network.py [hunk 4]
```diff
@@ -548,6 +539,214 @@ async def _race_with_page_close(self, future: Coroutine) -> None:
...
+                self._ws._channel.send(
+                    "close",
+                    {
+                        "code": code,
+                        "reason": reason,
+                    },
+                )
+            )
+        except Exception:
+            pass
```
Comment (dgozman) [inline playwright/_impl/_network.py:573]: Shouldn't you await the created task for this `try/except` to have an effect?

File: playwright/_impl/_network.py [hunk 5]
```diff
@@ -548,6 +539,214 @@ async def _race_with_page_close(self, future: Coroutine) -> None:
...
+
+    def send(self, message: Union[str, bytes]) -> None:
+        if isinstance(message, str):
+            asyncio.create_task(
```
Comment (dgozman) [inline playwright/_impl/_network.py:577]: Why not try/except here and below?

File: playwright/_impl/_tracing.py [hunk 6]
```diff
@@ -41,13 +41,10 @@ async def start(
...
         self._include_sources = bool(sources)
 
-        async def _inner_start() -> str:
-            await self._channel.send("tracingStart", params)
-            return await self._channel.send(
-                "tracingStartChunk", {"title": title, "name": name}
-            )
-
-        trace_name = await self._connection.wrap_api_call(_inner_start, True)
+        await self._channel.send("tracingStart", params)
```
Comment (dgozman) [inline playwright/_impl/_tracing.py:44]: I think there is a missing `markAsInternal()` call in this class.

File: tests/async/test_route_web_socket.py [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+
+import asyncio
+import re
+from typing import Any, Awaitable, Callable, Literal, Tuple, Union
+
+from playwright.async_api import Frame, Page, WebSocketRoute
+from tests.server import Server, WebSocketProtocol
+
+
+async def assert_equal(
```
Comment (dgozman) [inline tests/async/test_route_web_socket.py:23]: Looks like we need `expect.poll()` in python :)

File: tests/async/test_route_web_socket.py [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+    await page.evaluate("window.ws1.send('request')")
+    await assert_equal(lambda: page.evaluate("window.log"), ["ws1:response"])
+
+    await page.evaluate("window.ws2.send('request')")
+    await assert_equal(
+        lambda: page.evaluate("window.log"), ["ws1:response", "ws2:mock-response"]
+    )
+
+
+# test('should work with server', async ({ page, server }) => {
```
Comment (dgozman) [inline tests/async/test_route_web_socket.py:147]: Remove?

File: playwright/_impl/_fetch.py (hunk 3 above)
Comment (mxschmitt) [reply playwright/_impl/_fetch.py:408]: In Java we don't allow passing query params as strings while in JS/Python/.NET [we do](https://github.com/microsoft/playwright/pull/32516). There is no .NET roll yet for this release after we moved the merge logic to the [server side](https://github.com/microsoft/playwright/pull/32518).

In Java we have:
- https://playwright.dev/java/docs/api/class-requestoptions#request-options-set-query-param
- https://github.com/microsoft/playwright-java/blob/46f4ac1f33987ee3bc086e598d6d0ba197777087/playwright/src/main/java/com/microsoft/playwright/impl/RequestOptionsImpl.java

Comment (yury-s) [reply playwright/_impl/_fetch.py:408]: I'd rather not do it in the ports to keep the API surface smaller, especially if there is no user request for this.

Comment (Leoawad) [inline playwright/_impl/_fetch.py:408]: Perfect

**Pull request merged**